### PR TITLE
Add program listing to config file

### PR DESCRIPTION
**Pull request by dczulada**

### CODE REVIEW

**Comment by lmd59** [comment]
Potential issue: If a user submits a file and then goes "back" (<-) to the submission page, the Annual Update Bundle Section radio button remains selected, but the following sections are not shown.
screen shot 2017-07-14 at 2 48 13 pm
Potential solutions include deselecting all radio buttons when you return to the page or checking the status of buttons and populating subsequent sections accordingly.

**Comment by dczulada** [comment]
@lmd59 Updated to reset form when back is hit.

**Comment by lmd59** [comment]
"Back" fix looks good!
Additionally, I noticed that on radio button selection, we're initially sending back a 500 status before getting the correct information from the programs controller. This is something we could potentially fix up, though it doesn't have a strong effect on the user experience.
Error from console:
method=GET path=/programs/undefined format=html controller=ProgramsController action=programs_by_year status=500 duration=57.86 view=53.23
method=GET path=/programs/2017 format=html controller=ProgramsController action=programs_by_year status=200 duration=2.57 view=2.10

**Comment by dczulada** [comment]
Schematron's and voc.xml files have been verified.

**Comment by tstrass** [comment]
To be clear, I don't see anything wrong in the functionality. My comment is just in the interest of avoiding unnecessary complication in the app

File: app/views/uploads/new.html.erb [hunk 1]
```diff
@@ -1,226 +1,136 @@
...
+      <div class="form-group program-radio" id="<%= "radioContainer_#{program_year[0]}" %>" style="display: none;">
+          <%= render partial: 'program_radio', locals: {programs: program_year} %>
+      </div>
+    <% end %>
+  </fieldset>
+  <hr>
+  <fieldset>
+    <legend class="control-label">Document Type</legend>
+    <div class="form-group" id="radioDocumentContainer">
+    <% @document_years.each do |document_year| %>
```
Comment (tstrass) [inline app/views/uploads/new.html.erb]: Maybe I'm dense but it took me a while to figure out this loop and the one inside of it. Would you mind changing the iteration to using `|key, value|` syntax?
something like:
`<% @document_years.each do |year, document_types| %>`

and then something like this for the inner loop
`<% document_types.each do |category, specifications| %>`

File: app/views/uploads/new.html.erb [hunk 2]
```diff
@@ -1,226 +1,136 @@
...
+      <span id="bundle-message" class="help-block">To enable the <em>2016 Bundle for the 2017 Reporting Period</em>, please install a 2016 Cypress bundle</span>
+    <% end %>
+    <% unless BUNDLES['2018'] %>
+      <span id="bundle-message" class="help-block">To enable the <em>2017 Bundle for the 2018 Reporting Period</em>, please install a 2017 Cypress bundle</span>
+    <% end %>
+  </fieldset>
+  <hr>
+  <fieldset>
+    <legend class="control-label">CMS Reporting Program</legend>
+    <% @program_years.each do |program_year| %>
```
Comment (tstrass) [inline app/views/uploads/new.html.erb]: Can this be made into a `|key, value|` loop as well?

File: app/controllers/uploads_controller.rb [hunk 3]
```diff
@@ -7,13 +7,16 @@ def require_bundles
...
-    unless BUNDLES['2016'] || BUNDLES['2017']
-      raise "Please install a 2016 Cypress bundle or a 2017 Cypress bundle in order to use the Cypress Validation Utility"
+    unless BUNDLES['2016'] || BUNDLES['2017'] || BUNDLES['2018']
+      raise "Please install a 2016, 2017 or 2018 Cypress bundle in order to use the Cypress Validation Utility"
     end
   end
 
   def new
+    @program_years = YEAR_PROGRAMS
+    @document_years = YEAR_PROGRAMTYPE_DOCUMENTS
```
Comment (dczulada) [inline app/controllers/uploads_controller.rb]: Good idea

File: app/views/uploads/new.html.erb (hunk 2 above)
Comment (dczulada) [inline app/views/uploads/new.html.erb]: done

File: app/views/uploads/new.html.erb (hunk 1 above)
Comment (dczulada) [inline app/views/uploads/new.html.erb]: also done

**Pull request merged**